opslevel/resource_opslevel_check_alert_source_usage.go

```diff
@@ -1,86 +1,232 @@
+)
+
+func NewCheckAlertSourceUsageResource() resource.Resource {
+	return &CheckAlertSourceUsageResource{}
+}
+
+// CheckAlertSourceUsageResource defines the resource implementation.
+type CheckAlertSourceUsageResource struct {
+	CommonResourceClient
+}
+
+type CheckAlertSourceUsageResourceModel struct {
+	Category    types.String `tfsdk:"category"`
+	Description types.String `tfsdk:"description"`
+	Enabled     types.Bool   `tfsdk:"enabled"`
+	EnableOn    types.String `tfsdk:"enable_on"`
+	Filter      types.String `tfsdk:"filter"`
+	Id          types.String `tfsdk:"id"`
+	Level       types.String `tfsdk:"level"`
+	Name        types.String `tfsdk:"name"`
+	Notes       types.String `tfsdk:"notes"`
+	Owner       types.String `tfsdk:"owner"`
+	LastUpdated types.String `tfsdk:"last_updated"`
+
+	AlertType          types.String    `tfsdk:"alert_type"`
+	AlertNamePredicate *PredicateModel `tfsdk:"alert_name_predicate"`
+}
+
+func NewCheckAlertSourceUsageResourceModel(ctx context.Context, check opslevel.Check, planModel CheckAlertSourceUsageResourceModel) CheckAlertSourceUsageResourceModel {
+	var stateModel CheckAlertSourceUsageResourceModel
+
+	stateModel.Category = RequiredStringValue(string(check.Category.Id))
+	stateModel.Description = ComputedStringValue(check.Description)
+	if planModel.Enabled.IsNull() {
+		stateModel.Enabled = types.BoolValue(false)
+	} else {
+		stateModel.Enabled = OptionalBoolValue(&check.Enabled)
+	}
+	if planModel.EnableOn.IsNull() {
+		stateModel.EnableOn = types.StringNull()
+	} else {
+		// We pass through the plan value because of time formatting issue to ensure the state gets the exact value the customer specified
+		stateModel.EnableOn = planModel.EnableOn
+	}
+	stateModel.Filter = OptionalStringValue(string(check.Filter.Id))
+	stateModel.Id = ComputedStringValue(string(check.Id))
+	stateModel.Level = RequiredStringValue(string(check.Level.Id))
+	stateModel.Name = RequiredStringValue(check.Name)
+	stateModel.Notes = OptionalStringValue(check.Notes)
+	stateModel.Owner = OptionalStringValue(string(check.Owner.Team.Id))
+
+	stateModel.AlertType = types.StringValue(string(check.AlertSourceType))
+	if planModel.AlertNamePredicate != nil {
+		stateModel.AlertNamePredicate = NewPredicateModel(check.AlertSourceNamePredicate)
+	}
+
+	return stateModel
+}
+
+func (r *CheckAlertSourceUsageResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_check_alert_source_usage"
+}
+
+func (r *CheckAlertSourceUsageResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Check Alert Source Usage Resource",
+
+		Attributes: CheckBaseAttributes(map[string]schema.Attribute{
+			"alert_type": schema.StringAttribute{
+				Description: "",
```

Comment:
```suggestion
				Description: "The type of the alert source.",
```
From [docs](https://registry.terraform.io/providers/OpsLevel/opslevel/latest/docs/resources/check_alert_source_usage)